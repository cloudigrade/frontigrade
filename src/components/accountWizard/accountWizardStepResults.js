import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Spinner } from 'patternfly-react';
import { connect } from '../../redux';
import apiTypes from '../../constants/apiConstants';

const AccountWizardStepResults = ({ accountName, edit, error, errorMessage, fulfilled, pending }) => (
  <React.Fragment>
    {error && (
      <div className="wizard-pf-complete blank-slate-pf">
        <div className="wizard-pf-success-icon">
          <Icon type="pf" name="error-circle-o" />
        </div>
        <h3 className="blank-slate-pf-main-action">Error {edit ? 'Updating' : 'Creating'} Account</h3>
        <p className="blank-slate-pf-secondary-action">{errorMessage}</p>
      </div>
    )}
    {fulfilled && (
      <div className="wizard-pf-complete blank-slate-pf">
        <div className="wizard-pf-success-icon">
          <span className="glyphicon glyphicon-ok-circle" />
        </div>
        <h3 className="blank-slate-pf-main-action">
          <strong>{accountName}</strong> was {edit ? 'updated' : 'created'}.
        </h3>
      </div>
    )}
    {pending && (
      <div className="wizard-pf-process blank-slate-pf">
        <Spinner loading size="lg" className="blank-slate-pf-icon" />
        <h3 className="blank-slate-pf-main-action">{edit ? 'Updating' : 'Creating'} Account...</h3>
        <p className="blank-slate-pf-secondary-action">
          Please wait while account <strong>{accountName}</strong> is being {edit ? 'updated' : 'created'}.
        </p>
      </div>
    )}
  </React.Fragment>
);

AccountWizardStepResults.propTypes = {
  accountName: PropTypes.string,
  edit: PropTypes.bool,
  error: PropTypes.bool,
  errorMessage: PropTypes.string,
  fulfilled: PropTypes.bool,
  pending: PropTypes.bool
};

AccountWizardStepResults.defaultProps = {
  accountName: null,
  edit: false,
  error: false,
  errorMessage: null,
  fulfilled: false,
  pending: false
};

const mapStateToProps = state => ({
  accountName: state.accountWizard.account[apiTypes.API_ACCOUNT_NAME],
  ...state.accountWizard
});

const ConnectedAccountWizardStepResults = connect(mapStateToProps)(AccountWizardStepResults);

export { ConnectedAccountWizardStepResults as default, ConnectedAccountWizardStepResults, AccountWizardStepResults };
