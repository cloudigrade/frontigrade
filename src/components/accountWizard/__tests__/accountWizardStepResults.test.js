import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { ConnectedAccountWizardStepResults, AccountWizardStepResults } from '../accountWizardStepResults';

describe('AccountWizardStepResults Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a connected wizard results step with error', () => {
    const store = generateEmptyStore({
      accountWizard: { show: true, account: {}, error: true, errorMessage: 'Lorem ipsum' }
    });
    const component = shallow(<ConnectedAccountWizardStepResults />, { context: { store } });

    expect(component).toMatchSnapshot('connected');
  });

  it('should render a wizard results step with error', () => {
    const props = {
      edit: false,
      error: true,
      errorMessage: 'lorem ipsum'
    };

    const component = mount(<AccountWizardStepResults {...props} />);
    expect(component).toMatchSnapshot('error');

    component.setProps({ edit: true });
    expect(component).toMatchSnapshot('error edit');
  });

  it('should render a wizard results step with pending', () => {
    const props = {
      accountName: 'test',
      edit: false,
      pending: true
    };

    const component = mount(<AccountWizardStepResults {...props} />);
    expect(component).toMatchSnapshot('pending');

    component.setProps({ edit: true });
    expect(component).toMatchSnapshot('pending edit');
  });

  it('should render a wizard results step with fulfilled', () => {
    const props = {
      accountName: 'test',
      edit: false,
      fulfilled: true
    };

    const component = mount(<AccountWizardStepResults {...props} />);
    expect(component).toMatchSnapshot('fulfilled');

    component.setProps({ edit: true });
    expect(component).toMatchSnapshot('fulfilled edit');
  });
});
