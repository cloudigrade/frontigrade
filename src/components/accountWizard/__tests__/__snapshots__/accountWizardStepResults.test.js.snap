// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connected 1`] = `
<AccountWizardStepResults
  account={Object {}}
  accountName={null}
  dispatch={[Function]}
  edit={false}
  error={true}
  errorMessage="Lorem ipsum"
  fulfilled={false}
  pending={false}
  show={true}
/>
`;

exports[`error 1`] = `
<AccountWizardStepResults
  accountName={null}
  edit={false}
  error={true}
  errorMessage="lorem ipsum"
  fulfilled={false}
  pending={false}
>
  <div
    className="wizard-pf-complete blank-slate-pf"
  >
    <div
      className="wizard-pf-success-icon"
    >
      <Icon
        name="error-circle-o"
        type="pf"
      >
        <PatternflyIcon
          className=""
          name="error-circle-o"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-error-circle-o"
          />
        </PatternflyIcon>
      </Icon>
    </div>
    <h3
      className="blank-slate-pf-main-action"
    >
      Error 
      Creating
       Account
    </h3>
    <p
      className="blank-slate-pf-secondary-action"
    >
      lorem ipsum
    </p>
  </div>
</AccountWizardStepResults>
`;

exports[`error edit 1`] = `
<AccountWizardStepResults
  accountName={null}
  edit={true}
  error={true}
  errorMessage="lorem ipsum"
  fulfilled={false}
  pending={false}
>
  <div
    className="wizard-pf-complete blank-slate-pf"
  >
    <div
      className="wizard-pf-success-icon"
    >
      <Icon
        name="error-circle-o"
        type="pf"
      >
        <PatternflyIcon
          className=""
          name="error-circle-o"
        >
          <span
            aria-hidden="true"
            className="pficon pficon-error-circle-o"
          />
        </PatternflyIcon>
      </Icon>
    </div>
    <h3
      className="blank-slate-pf-main-action"
    >
      Error 
      Updating
       Account
    </h3>
    <p
      className="blank-slate-pf-secondary-action"
    >
      lorem ipsum
    </p>
  </div>
</AccountWizardStepResults>
`;

exports[`fulfilled 1`] = `
<AccountWizardStepResults
  accountName="test"
  edit={false}
  error={false}
  errorMessage={null}
  fulfilled={true}
  pending={false}
>
  <div
    className="wizard-pf-complete blank-slate-pf"
  >
    <div
      className="wizard-pf-success-icon"
    >
      <span
        className="glyphicon glyphicon-ok-circle"
      />
    </div>
    <h3
      className="blank-slate-pf-main-action"
    >
      <strong>
        test
      </strong>
       was 
      created
      .
    </h3>
  </div>
</AccountWizardStepResults>
`;

exports[`fulfilled edit 1`] = `
<AccountWizardStepResults
  accountName="test"
  edit={true}
  error={false}
  errorMessage={null}
  fulfilled={true}
  pending={false}
>
  <div
    className="wizard-pf-complete blank-slate-pf"
  >
    <div
      className="wizard-pf-success-icon"
    >
      <span
        className="glyphicon glyphicon-ok-circle"
      />
    </div>
    <h3
      className="blank-slate-pf-main-action"
    >
      <strong>
        test
      </strong>
       was 
      updated
      .
    </h3>
  </div>
</AccountWizardStepResults>
`;

exports[`pending 1`] = `
<AccountWizardStepResults
  accountName="test"
  edit={false}
  error={false}
  errorMessage={null}
  fulfilled={false}
  pending={true}
>
  <div
    className="wizard-pf-process blank-slate-pf"
  >
    <Spinner
      className="blank-slate-pf-icon"
      inline={false}
      inverse={false}
      loading={true}
      size="lg"
    >
      <div
        className="blank-slate-pf-icon spinner spinner-lg"
      />
    </Spinner>
    <h3
      className="blank-slate-pf-main-action"
    >
      Creating
       Account...
    </h3>
    <p
      className="blank-slate-pf-secondary-action"
    >
      Please wait while account 
      <strong>
        test
      </strong>
       is being 
      created
      .
    </p>
  </div>
</AccountWizardStepResults>
`;

exports[`pending edit 1`] = `
<AccountWizardStepResults
  accountName="test"
  edit={true}
  error={false}
  errorMessage={null}
  fulfilled={false}
  pending={true}
>
  <div
    className="wizard-pf-process blank-slate-pf"
  >
    <Spinner
      className="blank-slate-pf-icon"
      inline={false}
      inverse={false}
      loading={true}
      size="lg"
    >
      <div
        className="blank-slate-pf-icon spinner spinner-lg"
      />
    </Spinner>
    <h3
      className="blank-slate-pf-main-action"
    >
      Updating
       Account...
    </h3>
    <p
      className="blank-slate-pf-secondary-action"
    >
      Please wait while account 
      <strong>
        test
      </strong>
       is being 
      updated
      .
    </p>
  </div>
</AccountWizardStepResults>
`;
